src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/SetAzureVpnClientIpsecParametersCommand.cs

```diff
@@ -0,0 +1,76 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.Network.VirtualNetworkGateway;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Network;
+using System;
+using System.Management.Automation;
+using MNM = Microsoft.Azure.Management.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Set,
+         "AzureRmVpnClientIpsecParameter",
+         DefaultParameterSetName = VirtualNetworkGatewayParameterSets.Default,
+         SupportsShouldProcess = true),
+     OutputType(typeof(PSVpnClientIPsecParameters))]
+    public class SetAzureVpnClientIpsecParametersCommand : VirtualNetworkGatewayBaseCmdlet
```

Comment:
@Nilambari this cmdlet should allow users to provide an `InputObject` or `ResourceId` parameter (similar to what was done for `Remove-*`)